LoopKitUI/Views/FractionalQuantityPicker.swift

```diff
@@ -148,6 +151,81 @@ struct FractionalQuantityPicker: View {
         }
     }
 
+    private var wholeGuardrail: Guardrail<HKQuantity> {
+        // Whether or not the whole value should be colored as 'out of range' depends on the fraction value
+        let lowerRecommendedBound = guardrail.recommendedBounds.lowerBound.doubleValue(for: unit)
+        if fraction < lowerRecommendedBound.fraction {
+            return guardrail
+        } else {
+            // Shift the lower bounds down one 'whole' value
+            return Guardrail(
+                absoluteBounds: HKQuantity(unit: unit, doubleValue: guardrail.absoluteBounds.lowerBound.doubleValue(for: unit) - 1)...guardrail.absoluteBounds.upperBound,
+                recommendedBounds: HKQuantity(unit: unit, doubleValue: lowerRecommendedBound - 1)...guardrail.recommendedBounds.upperBound
```

Comment:
[nit] I think I'm missing something. I find this confusing. Possible more details in the comment may help.

I'm confused because it reads like if the fractional part of the picker is >= the lower recommended bound fractional part (without considering the whole part) then -1 from the lower recommended bound and lower absolute bounds. I don't understand why this is the outcome.

Also, what if the lower absolute bounds was already 0.x. Would this not provide a negative lower absolute bounds?

Reply:
I'll flesh out the comment more.

Yes, it would be provide a negative lower absolute bounds, but the selectable values aren't determined by the guardrail, just the warning colors. If the absolute lower bound is smaller than the smallest selectable value, then none of the selectable values will be highlighted with the red 'minimum' color.